test/integration/FileResourceStore.test.ts

```diff
@@ -0,0 +1,151 @@
+import { promises as fs } from 'fs';
+import * as url from 'url';
+import { MockResponse } from 'node-mocks-http';
+import { FileResourceStoreConfig } from '../configs/FileResourceStoreConfig';
+import { call, callFile } from '../util/Util';
+
+describe('A server using a FileResourceStore', (): void => {
+  describe('without acl', (): void => {
+    const config = new FileResourceStoreConfig();
+    const handler = config.getHandler();
+
+    it('can add a file to the store, read it and delete it.', async(): Promise<
+    void
+    > => {
+      // POST
+      let requestUrl = new URL('http://test.com/');
+
+      const fileData = await fs.readFile('test/testfiles/testfile1.txt');
+
+      let response: MockResponse<any> = await callFile(
+        handler,
+        requestUrl,
+        'POST',
+        { 'content-type': 'application/octet-stream', slug: 'testfile1.txt', 'transfer-encoding': 'chunked' },
+        fileData,
+      );
+
+      expect(response.statusCode).toBe(200);
+      expect(response._getData()).toHaveLength(0);
+      const id = response._getHeaders().location;
+      expect(id).toContain(url.format(requestUrl));
+
+      // GET
+      requestUrl = new URL(id);
+      response = await call(handler, requestUrl, 'GET', { accept: 'text/*' }, []);
+      expect(response.statusCode).toBe(200);
+      expect(response._getHeaders().location).toBe(id);
+      expect(response._getBuffer().toString()).toContain('TESTFILE1');
+
+      // DELETE
+      response = await call(handler, requestUrl, 'DELETE', {}, []);
+      expect(response.statusCode).toBe(200);
+      expect(response._getData()).toHaveLength(0);
+      expect(response._getHeaders().location).toBe(url.format(requestUrl));
+
+      // GET
+      response = await call(handler, requestUrl, 'GET', { accept: 'text/*' }, []);
+      expect(response.statusCode).toBe(404);
+      expect(response._getData()).toContain('NotFoundHttpError');
+    });
+
+    it('can add and overwrite a file.', async(): Promise<
+    void
+    > => {
+      // POST
+      let requestUrl = new URL('http://test.com/');
+
+      let fileData = await fs.readFile('test/testfiles/testfile1.txt');
+
+      let response: MockResponse<any> = await callFile(
+        handler,
+        requestUrl,
+        'POST',
+        { 'content-type': 'application/octet-stream', slug: 'testfile1.txt', 'transfer-encoding': 'chunked' },
+        fileData,
+      );
+
+      expect(response.statusCode).toBe(200);
+      expect(response._getData()).toHaveLength(0);
+      const id = response._getHeaders().location;
+      expect(id).toContain(url.format(requestUrl));
+
+      // GET
+      requestUrl = new URL(id);
+      response = await call(handler, requestUrl, 'GET', { accept: 'text/*' }, []);
+      expect(response.statusCode).toBe(200);
+      expect(response._getHeaders().location).toBe(id);
+      expect(response._getBuffer().toString()).toContain('TESTFILE1');
```

Comment:
A succesful GET is always going to have these 5 lines right? So I'm wondering if this can be reduced to a single line calling a function that covers this (still with the possibility of extending it later for calls that have more specifics)? Similarly for all the other HTTP methods. This is just musing so it could be that there is no clean way to do it.

Reply:
This is actually what we still wanted to do, as there is a lot of similar code (same with the other requests & acl). We might have some ideas that hopefully can get worked out before Friday.

Reply:
I don't mind being explicit for a unit test.

Reply:
True but in all integrationtests are these lines multiple times duplicated. Some form of abstraction could make is easier to read and write these tests.

Reply:
Anything in https://github.com/jest-community/awesome-jest#matchers we could leverage?

Reply:
Or also `supertest` which we are using elsewhere:

```
  const server = supertest(serverUri)

  describe('Unauthenticated requests to protected resources', () => {
    describe('accepting text/html', () => {
      it('should return 401 Unauthorized with www-auth header', () => {
        return server.get('/profile/')
          .set('Accept', 'text/html')
          .expect('WWW-Authenticate', 'Bearer realm="https://localhost:3457", scope="openid webid"')
          .expect(401)
      })
```

Reply:
hmmm we were thinking something like supertest. So its easier to do calls for the different methods. Something like: `callHelper.get(url, headers, statusCode)` and have a check for statusCode and the header and than return the `response` so extra checks like .toContain could be done to have more freedom (it seems not possible with supertest).

Reply:
Noticed that supertest does return the response, it seems like we have three options now: 

Either we keep testing the handler with the util call functions and make them shorter ourselves by also doing some generic checks in a `callHelper.get(..)` or even `callHelper.successfulGet(..)` like Thor & Joachim implied.

We could also move the tests to work with supertest and wrap the handler in a httpServer. Would still have to see if that translates to the tests we have now. Also not sure if supertest would really help with making this much shorter. (I think this would also take some more time as it is a bigger change to the code).

Or another option: keeping the tests/checks explicit but just shortening the call function (by doing as Joachim said in a comment above: a new function where the fixed parameters such as handler are already set). We could maybe even separate these call functions into get, post, ... but just not do any checks there (while option 1 would already do some checks I guess).

Reply:
I also don't think supertest is going to make it shorter, but it might still be interesting to see if it can be used if that would mean the http handler can be included in the integration tests (but would then still see if some parts can be shortened by using function calls). Would not spend too much time on this now if it would require a complete refactor though.